libmount: fix statx() includes

karelzak opened the pull request:
Using sys/stat.h and linux/stat is too tricky.h together. It seems better to rely on libc and use sys/stat.h only. Users affected by old libc must update to use recent util-linux.

Fixes: https://github.com/util-linux/util-linux/issues/2448

nolange commented:
Compiles now, did not do any functional tests.

`linux/stat.h` could be now removed from `AC_CHECK_HEADERS`

Not related to this issue, I don't see `EINVAL` handled, which would happen if you run the binary on an older kernel (while compiled with newer headers). Should util-linux compiled for recent kernels work correctly when run on older kernels?

karelzak commented:
> Compiles now, did not do any functional tests.

Cool.

> `linux/stat.h` could be now removed from `AC_CHECK_HEADERS`

Good catch; I also forgot to update the meson stuff. Fixed now.

> Not related to this issue, I don't see `EINVAL` handled, which would happen if you run the binary on an older kernel (while compiled with newer headers). Should util-linux compiled for recent kernels work correctly when run on older kernels?

Yes, the code seems fragile (it was originally just a hack to see if we could get the mount ID from the not-yet-attached mount node). Fixed now. 

Thanks!
